src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -137,6 +137,9 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                 {
                     await RestoreAsync(request.ForceRestore, request.RestoreSource, token);
                 }
+                catch (OperationCanceledException) when (token.IsCancellationRequested)
```

Comment:
Why do you swallow `OperationCanceledException` when cancellation is requested? The contract is that if a token is canceled but the method completes without throwing that exception, it must have completed the operation. If the user cancels and you abort without completing, you should throw the exception so your caller knows it was not completed.